lib/functions.php

```diff
@@ -404,11 +404,9 @@ function decodePath(string $path): string
 function decodePathSegment(string $path): string
 {
     $path = rawurldecode($path);
-    $encoding = mb_detect_encoding($path, ['UTF-8', 'ISO-8859-1']);
 
-    switch ($encoding) {
-        case 'ISO-8859-1':
-            $path = utf8_encode($path);
```

Comment:
since this method is [deprecated as of php 8.2](https://wiki.php.net/rfc/remove_utf8_decode_and_utf8_encode), this is a step into a good direction

:+1: